src/compiler/program.ts

```diff
@@ -1520,7 +1527,297 @@ namespace ts {
+                    const checker = getTypeChecker();
+                    startExtensionProfile(profileLevel, name, "construct", host.trace);
+                    try {
+                        walker = new (ctor as SemanticLintProviderStatic)({ ts, checker, args, host, program });
+                    }
+                    catch (e) {
+                        diagnostics.push(createExtensionDiagnostic(name, `Lint construction failed: ${(e as Error).message}`, sourceFile, /*start*/undefined, /*length*/undefined, DiagnosticCategory.Error));
+                    }
+                    completeExtensionProfile(profileLevel, name, "construct", host.trace);
+                }
+                else if (kind === ExtensionKind.SyntacticLint) {
+                    startExtensionProfile(profileLevel, name, "construct", host.trace);
+                    try {
+                        walker = new (ctor as SyntacticLintProviderStatic)({ ts, args, host, program });
+                    }
+                    catch (e) {
+                        diagnostics.push(createExtensionDiagnostic(name, `Lint construction failed: ${(e as Error).message}`, /*sourceFile*/undefined, /*start*/undefined, /*length*/undefined, DiagnosticCategory.Error));
+                    }
+                    completeExtensionProfile(profileLevel, name, "construct", host.trace);
+                }
+                if (walker) initializedLints.push({ name, walker, accepted: true, errored: false });
+            }
+
+            let nodesVisited = 0;
+            visitNode(sourceFile);
+
+            return diagnostics;
+
+            function visitNode(node: Node) {
+                node.parent = parent;
+                nodesVisited++;
+                let oneAccepted = false;
+                const oldParent = parent;
+                const needsReset: Map<boolean> = {};
+                for (let i = 0; i < initializedLints.length; i++) {
+                    activeLint = initializedLints[i];
+                    if (activeLint.accepted) {
+                        if (!activeLint.errored) {
+                            startExtensionProfile(profileLevel, activeLint.name, `visitNode|${nodesVisited}`, host.trace);
+                            try {
+                                activeLint.accepted = !activeLint.walker.visit(node, error);
+                            }
+                            catch (e) {
+                                activeLint.errored = true;
+                                diagnostics.push(createExtensionDiagnostic(errorQualifiedName("!!!"), `visit failed with error: ${e}`, /*sourceFile*/undefined, /*start*/undefined, /*length*/undefined, DiagnosticCategory.Error));
+                            }
+                            completeExtensionProfile(profileLevel, activeLint.name, `visitNode|${nodesVisited}`, host.trace);
+                        }
+                        if (activeLint.accepted) {

```

Comment:
I just need to move this block into the `!activeLint.errored` block above, then set `activeLint.accepted = false` when I set errored. It wasn't calling the rule anymore, but you're right - no need to keep recuring on account of a failed rule.